src/tree/taffy_tree/tree.rs

```diff
@@ -57,22 +63,48 @@ impl Default for Taffy {
     }
 }
 
-/// Iterator that wraps a slice of nodes, lazily converting them to u64
-pub struct TaffyChildIter<'a>(core::slice::Iter<'a, NodeId>);
-impl<'a> Iterator for TaffyChildIter<'a> {
+/// Iterator over the Vec in a RefMut<'a, Vec<NodeId>>
+pub struct RefCellVecIter<'a> {
+    /// The items to iterate over
+    children: RefMut<'a, Vec<NodeId>>,
+    /// The next index to return
+    index: usize,
+}
+impl<'a> Iterator for RefCellVecIter<'a> {
     type Item = NodeId;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.0.next().copied()
+        let item = self.children.get(self.index).copied();
+        self.index += 1;
+        item
+    }
+}
+
+/// Iterates over children, checking the Display type of the node
+/// If the node is `Display::Contents`, then we recurse it's children, else we simply push the `NodeId` into the list
+fn find_children_recursive(out: &mut Vec<NodeId>, tree: &Taffy, node: NodeId) {
+    for child_id in tree.children[node.into()].iter() {
+        let child_key: DefaultKey = (*child_id).into();
+        let display = tree.nodes[child_key].style.display;
+        match display {
+            Display::Contents => find_children_recursive(out, tree, *child_id),
+            _ => out.push(*child_id),
+        }
     }
 }
 
 impl LayoutTree for Taffy {
-    type ChildIter<'a> = TaffyChildIter<'a>;
+    type ChildIter<'a> = RefCellVecIter<'a>;
 
     #[inline(always)]
     fn children(&self, node: NodeId) -> Self::ChildIter<'_> {
-        TaffyChildIter(self.children[node.into()].iter())
+        let mut cache = self.node_children_cache.borrow_mut();
+        if cache.0 != node {
+            cache.1.clear();
+            cache.0 = node;
+            find_children_recursive(&mut cache.1, self, node);
```

Comment:
This section is a bit hard to read, maybe with a struct with named properties it would be easier?

Reply:
Good shout. I've changed this to use a struct :)